pkg/s3select/funcEval.go

```diff
@@ -140,26 +138,54 @@ func evaluateFuncExpr(myVal *sqlparser.FuncExpr, myReturnVal string, myRecord []
 	for i := 0; i < len(mySubFunc); i++ {
 		if supportedString(myVal.Name.CompliantName()) {
 			if mySubFunc != nil {
-				return stringOps(myVal, myRecord, evaluateFuncExpr(mySubFunc[i], myReturnVal, myRecord, columnsMap), columnsMap)
+				return stringOps(myVal, myRecord, evaluateFuncExpr(mySubFunc[i], myReturnVal, myRecord))
 			}
-			return stringOps(myVal, myRecord, myReturnVal, columnsMap)
+			return stringOps(myVal, myRecord, myReturnVal)
 		} else if strings.ToUpper(myVal.Name.CompliantName()) == "NULLIF" {
 			if mySubFunc != nil {
-				return nullOps(myVal, myRecord, evaluateFuncExpr(mySubFunc[i], myReturnVal, myRecord, columnsMap), columnsMap)
+				return nullOps(myVal, myRecord, evaluateFuncExpr(mySubFunc[i], myReturnVal, myRecord))
 			}
-			return nullOps(myVal, myRecord, myReturnVal, columnsMap)
+			return nullOps(myVal, myRecord, myReturnVal)
 		} else if strings.ToUpper(myVal.Name.CompliantName()) == "COALESCE" {
 			if mySubFunc != nil {
-				return coalOps(myVal, myRecord, evaluateFuncExpr(mySubFunc[i], myReturnVal, myRecord, columnsMap), columnsMap)
+				return coalOps(myVal, myRecord, evaluateFuncExpr(mySubFunc[i], myReturnVal, myRecord))
 			}
-			return coalOps(myVal, myRecord, myReturnVal, columnsMap)
+			return coalOps(myVal, myRecord, myReturnVal)
 		}
 	}
 	return ""
 }
 
 // evaluateFuncErr is a function that flags errors in nested functions.
-func (reader *Input) evaluateFuncErr(myVal *sqlparser.FuncExpr) error {
+func (reader *CSVInput) evaluateFuncErr(myVal *sqlparser.FuncExpr) error {
```

Comment:
this function and the one following are both doing the same thing. Can the function be changed to `func evaluateFuncErr(myVal *sqlparser.FuncExpr, reader *SelectQuery) error`

Reply:
Done.